CMakeLists.txt

```diff
@@ -66,16 +66,22 @@ function(use_faster_linker)
     endforeach()
 endfunction()
 
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_SYSTEM_PROCESSOR STREQUAL "arm")
+    set(REALM_RPI ON)
+endif()
+
 # Set global warnings settings
 if(MSVC)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W3 /D_CRT_SECURE_NO_WARNINGS /D_SCL_SECURE_NO_WARNINGS")
 else()
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Wundef -Wshadow")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Wundef -Wshadow -D_FILE_OFFSET_BITS=64")
```

Comment:
I don't want to only do this on GLIBC since it seems to be needed on android as well. I think it is needed for any linux build, and possibly also solaris. I made it only define it if `sizeof(off_t) == 4`.

Reply:
Oh, my understanding is that `_FILE_OFFSET_BITS` is only a thing on the GNU libc. If that's not the case then you have a good solution. Also, nice catch.